src/main/java/Event.java

```diff
@@ -0,0 +1,16 @@
+import java.util.ArrayList;
+
+// Event class representing an event, with fields course, teacher, date and time
+public class Event {
+    public Event(Teacher teacher, String date, String time) {
+        this.teacher = teacher;
+        this.date = date;
+        this.time = time;
+    }
+    
+    Course course;
+    Teacher teacher;
```

Comment:
Since course already knows about the teacher, wouldn't this pointer make things redundant ?

Reply:
done